### PR TITLE
GHC-8.2

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/Data/Aeson/Extra/Merge.hs [hunk 1]
```diff
@@ -51,3 +39,21 @@ mergeA :: Functor f
...
       -> Value -> Value -> f Value
 mergeA f a b = embed <$> f (mergeA f) (project a) (project b)
+
+-- | Example of using 'merge'. see <https://lodash.com/docs#merge>:
+--
+-- /Note:/ not tested against JavaScript lodash, so may disagree in the results.
+--
+-- @since 0.4.1.0
+lodashMerge :: Value -> Value -> Value
+lodashMerge = merge alg
```
Comment (Profpatsch) [inline src/Data/Aeson/Extra/Merge.hs]: I’d call it `rightBiasedMerge` and skip the lodash association. Maybe mention that it should behave like lodash, but is not tested against it.

Comment (phadej) [inline src/Data/Aeson/Extra/Merge.hs]: `rightBiasedMerge` is undescriptive as well. I'd rather stay with `lodashMerge` and rather document differences as we learn them.

Comment (Profpatsch) [inline src/Data/Aeson/Extra/Merge.hs]: Well, the description on the lodash site contains the case on `undefined`; also JS libraries like this are normally everything but short lived (and links to their APIs even more), so I’d not introduce a dependency on that for such a simple merge algrorithm.

```haskell
-- | A simple merge, combining Values recursively and defaulting to the right hand side if it is of a different constructor.
--
-- Pseudo-code:
-- @
-- rightBiasedMerge { a: { b: 3, c: 4 } } { a: { c: 5, d: 6 } } == { a: { b: 3, c: 5, d: 6 } }
-- rightBiasedMerge [ 1 2 3 ] 4 == 4
-- rightBiasedMerge { a: 3 } [ 1 2 3 ] == [ 1 2 3 ]
rightBiasedMerge :: …
```

Comment (phadej) [inline src/Data/Aeson/Extra/Merge.hs]: Lodash is most popular library on https://www.npmjs.com/, and the docs link was the same from the moment `Merge` module was introduced here (1.5 years ago). 

Please, don't generalize. Bashing JS won't help to change my mind.

**Pull request merged**